signer/src/dkg/verification.rs

```diff
@@ -0,0 +1,766 @@
+            .expect("should be able to process message");
+        assert_state!(state_machine, State::Signing);
+        assert_allowed_msg_type!(state_machine, NonceResponse);
+        state_machine
+            .process_message(sender2, nonce_response2.clone())
+            .expect("should be able to process message");
+        assert_state!(state_machine, State::Signing);
+
+        // We should have processed all the nonce responses, so we should now be
+        // able to process signature share requests.
+        assert_allowed_msg_type!(state_machine, SignatureShareResponse);
+    }
+
+    #[test]
+    fn test_dkg_verification_state_machine() {
+        let mut setup = TestSetup::setup(2);
+        let sender1 = setup.sender(0);
+        let sender2 = setup.sender(1);
+        let mut state_machine = setup.state_machine;
+        let mut signer1 = setup.signers.pop().unwrap();
+        let mut signer2 = setup.signers.pop().unwrap();
+
+        let nonce_request = nonce_request(1, 1, 1);
+
+        // Process the nonce request with signer 1 and 2 to get their responses.
+        let nonce_response1 = signer1.process(&nonce_request).unwrap().single();
+        let nonce_response2 = signer2.process(&nonce_request).unwrap().single();
+
+        // Process the nonce request in the state machine and assert.
+        state_machine
+            .process_message(sender1, nonce_request)
+            .unwrap();
+        assert!(state_machine.message_counts(NonceRequest, 1, 0));
+        assert!(matches!(state_machine.state(), &State::Signing));
+        assert!(matches!(
+            state_machine.coordinator.state,
+            wsts::state_machine::coordinator::State::NonceGather(_)
+        ));
+
+        // Process both nonce responses in the state machine and assert.
+        state_machine
+            .process_message(sender1, nonce_response1.clone())
+            .expect("should be able to process message");
+        assert_state!(state_machine, State::Signing);
+        assert_allowed_msg_type!(state_machine, NonceResponse);
+        state_machine
+            .process_message(sender2, nonce_response2.clone())
+            .expect("should be able to process message");
+        assert_state!(state_machine, State::Signing);
+        assert_allowed_msg_type!(state_machine, SignatureShareResponse);
+
+        // Unwrap our nonce responses.
+        let Message::NonceResponse(nonce_response1) = nonce_response1 else {
+            panic!("expected nonce response, got {:?}", nonce_response1);
+        };
+        let Message::NonceResponse(nonce_response2) = nonce_response2 else {
+            panic!("expected nonce response, got {:?}", nonce_response2);
+        };
+
+        // Create signature share requests, populated with the nonce responses
+        // from the signers.
+        let sig_share_request = signature_share_request(
+            1,
+            1,
+            1,
+            vec![nonce_response2.clone(), nonce_response1.clone()],
```

Comment:
If you leave the `pop` call above, please comment here why the reverse ordering is necessary.

Reply:
Removed pop: [75c78bc](https://github.com/stacks-network/sbtc/pull/1324/commits/75c78bc3c46868836857bc3750671ccbec4c83c3)